pkgs/development/python-modules/plyer/default.nix

```diff
@@ -0,0 +1,60 @@
+{ lib, buildPythonPackage, fetchFromGitHub, fetchpatch, keyring, mock, pytestCheckHook, stdenv }:
+
+buildPythonPackage rec {
+  pname = "plyer";
+  version = "2.0.0";
+
+  src = fetchFromGitHub {
+    owner = "kivy";
+    repo = "plyer";
+    rev = version;
+    sha256 = "15z1wpq6s69s76r6akzgg340bpc21l2r1j8270gp7i1rpnffcjwm";
+  };
+
+  patches = [
+    # fix naming of the DOCUMENTS dir
+    (fetchpatch {
+      url = "https://github.com/rski/plyer/commit/99dabb2d62248fc3ea5705c2720abf71c9fc378b.patch";
+      sha256 = "1ygahxpb8l6bqwpfba0cm164mwyc2snhpdighzyw540s1l0n31g8";
+    })
+    # fix handling of the ~/.config/user-dirs.dir file
+    (fetchpatch {
+      url = "https://github.com/rski/plyer/commit/f803697a1fe4fb5e9c729ee6ef1997b8d64f3ccd.patch";
+      sha256 = "1ciksqfn63bpn9p3s1min24ipjqp32pgcsffabkxj2b5d1nnbplz";
+    })
+  ];
+
+  postPatch = ''
+    rm -r examples
+    # remove all the wifi stuff. Depends on a python wifi module that has not been updated since 2016
+    find -iname "wifi*" -exec rm {} \;
+    substituteInPlace plyer/__init__.py \
+      --replace "wifi = Proxy('wifi', facades.Wifi)" "" \
+      --replace "'wifi'" ""
+    substituteInPlace plyer/facades/__init__.py \
+      --replace "from plyer.facades.wifi import Wifi" ""
+  '';
+
+  propagatedBuildInputs = [ keyring ];
+
+  checkInputs = [ mock pytestCheckHook ];
+
+  # notifysend assumes dbus is not installed, it fails and is not very robust.
+  # test_uniqueid fails during nix-build, but I am not able to explain why. 
+  # The test and the API under test do work outside the nix build.
+  pytestFlagsArray = [ "plyer/tests" ''-k "not test_notification_notifysend and not test_uniqueid"'' ];
```

Comment:
You should be able to use disabledTests instead of -k in pytestFlagsArray

Reply:
oh nice, done